kernel/kernel-api/src/main/java/io/delta/kernel/internal/files/ParsedLogData.java

```diff
@@ -0,0 +1,200 @@
+ */
+public class ParsedLogData {
+
+  ///////////////////////////////////////
+  // Static enums, fields, and methods //
+  ///////////////////////////////////////
+
+  public enum ParsedLogCategory {
+    DELTA,
+    LOG_COMPACTION,
+    CHECKPOINT,
+    CHECKSUM
+  }
+
+  public enum ParsedLogType {
+    PUBLISHED_DELTA(ParsedLogCategory.DELTA),
+    RATIFIED_STAGED_COMMIT(ParsedLogCategory.DELTA),
+    RATIFIED_INLINE_COMMIT(ParsedLogCategory.DELTA),
+    LOG_COMPACTION(ParsedLogCategory.LOG_COMPACTION),
+    CLASSIC_CHECKPOINT(ParsedLogCategory.CHECKPOINT),
+    MULTIPART_CHECKPOINT(ParsedLogCategory.CHECKPOINT),
+    V2_CHECKPOINT(ParsedLogCategory.CHECKPOINT),
+    CHECKSUM(ParsedLogCategory.CHECKSUM);
+
+    public final ParsedLogCategory category;
+
+    ParsedLogType(ParsedLogCategory category) {
+      this.category = category;
+    }
+  }
+
+  public static ParsedLogData forFileStatus(FileStatus fileStatus) {
+    final String path = fileStatus.getPath();
+
+    if (FileNames.isLogCompactionFile(path)) {
+      return ParsedLogCompactionData.forFileStatus(fileStatus);
+    } else if (FileNames.isCheckpointFile(path)) {
+      return ParsedCheckpointData.forFileStatus(fileStatus);
+    }
+
+    final long version;
+    final ParsedLogType type;
+
+    if (FileNames.isPublishedDeltaFile(path)) {
+      version = FileNames.deltaVersion(path);
+      type = ParsedLogType.PUBLISHED_DELTA;
+    } else if (FileNames.isStagedDeltaFile(path)) {
+      version = FileNames.deltaVersion(path);
+      type = ParsedLogType.RATIFIED_STAGED_COMMIT;
+    } else if (FileNames.isChecksumFile(path)) {
+      version = FileNames.checksumVersion(path);
+      type = ParsedLogType.CHECKSUM;
+    } else {
+      throw new IllegalArgumentException("Unknown log file type: " + path);
+    }
+
+    return new ParsedLogData(version, type, Optional.of(fileStatus), Optional.empty());
+  }
+
+  public static ParsedLogData forInlineData(
+      long version, ParsedLogType type, ColumnarBatch inlineData) {
+    if (type == ParsedLogType.PUBLISHED_DELTA || type == ParsedLogType.RATIFIED_STAGED_COMMIT) {
+      throw new IllegalArgumentException(
+          "For PUBLISHED_DELTA|RATIFIED_STAGED_COMMIT, use ParsedLogData.forFileStatus() instead");
+    } else if (type == ParsedLogType.LOG_COMPACTION) {
+      throw new IllegalArgumentException(
+          "For LOG_COMPACTION, use ParsedLogCompactionData.forInlineData() instead");
+    } else if (type.category == ParsedLogCategory.CHECKPOINT) {
+      return ParsedCheckpointData.forInlineData(version, type, inlineData);
+    }
+    return new ParsedLogData(version, type, Optional.empty(), Optional.of(inlineData));
+  }
+
+  ///////////////////////////////
+  // Member fields and methods //
+  ///////////////////////////////
+
+  public final long version;
+  public final ParsedLogType type;
+  public final Optional<FileStatus> fileStatusOpt;
+  public final Optional<ColumnarBatch> inlineDataOpt;
```

Comment:
Just clarify: the assumption inline data could only contain 1 batch, however, there could be multiple rows in the batch?

Reply:
Yup -- it's just 1 columnarBatch (not a list of columnarBatch) -- and yes that columnarBatch can contain many rows